compose-lint-checks/src/main/java/slack/lint/compose/ViewModelInjectionDetector.kt

```diff
@@ -8,15 +8,28 @@ import com.android.tools.lint.detector.api.Issue
 import com.android.tools.lint.detector.api.JavaContext
 import com.android.tools.lint.detector.api.Severity
 import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.StringOption
 import org.jetbrains.kotlin.psi.KtCallExpression
 import org.jetbrains.kotlin.psi.KtFunction
 import org.jetbrains.kotlin.psi.KtProperty
 import slack.lint.compose.util.*
 import slack.lint.compose.util.sourceImplementation
 
-class ViewModelInjectionDetector : ComposableFunctionDetector(), SourceCodeScanner {
+class ViewModelInjectionDetector
+@JvmOverloads
+constructor(private val allowedNames: StringSetLintOption = StringSetLintOption(ALLOW_LIST)) :
```

Comment:
Went back and forth a few times if this should be a `LintOption` or a `StringSetLintOption` and ended up leaving it as a `StringSetLintOption` for scenarios where codebases might be transitioning between ViewModel injection methods.

Reply:
For my own understanding – what would the reason to use a `LintOption` be?

Reply:
Oops! For some reason in my head `LintOption` had a base implementation that contained a single String instead of a Set of Strings. If that was actually the case I was trying to decide if the more common use case would be for a single name being set or multiple names being set.